guidance/guidance_los/guidance_los/los_guidance_algorithm.py

```diff
@@ -0,0 +1,138 @@
+# integrated_los_guidance.py
+# location: vortex-auv/guidance/guidance_los/guidance_los/los_guidance_algorithm.py
+
+#!/usr/bin/env python3
+
+import numpy as np
+from typing import Tuple
+
+class FirstOrderLOSGuidance:
+    def __init__(self):
+        # LOS parameters (keeping same as original)
+        self.h_delta_min = 1.0      # Minimum look-ahead distance
+        self.h_delta_max = 5.0      # Maximum look-ahead distance
+        self.h_delta_factor = 3.0   # Look-ahead scaling factor
+        self.nominal_speed = 0.35   # Nominal surge velocity
+        self.min_speed = 0.1        # Minimum surge velocity
+        self.max_pitch_angle = np.pi/6  # 30 degrees max pitch
+        self.depth_gain = 0.5       # Gain for depth error
+
+        # First order filter parameters
+        self.x = np.zeros(3)  # State vector for reference filter [surge, pitch, yaw]
+        self.omega = np.diag([2.5, 2.5, 2.5])  # Natural frequency - can be tuned
+        
+        # Initialize filter matrices
+        self.setup_filter_matrices()
+
+    def setup_filter_matrices(self):
+        """Setup state-space matrices for the first-order reference filter."""
+        # Simple first order filter matrices
+        self.Ad = -self.omega
+        self.Bd = self.omega
+
+    def normalize_angle(self, angle: float) -> float:
+        """Normalize angle to [-pi, pi]."""
+        return (angle + np.pi) % (2 * np.pi) - np.pi
+
+    def compute_raw_los_guidance(self, current_pos: np.ndarray, 
+                               target_pos: np.ndarray) -> Tuple[np.ndarray, float, float]:
+        """
+        Compute raw LOS guidance commands.
+        
+        Args:
+            current_pos: np.ndarray [x, y, z, yaw, pitch]
+            target_pos: np.ndarray [x, y, z]
+            
+        Returns:
+            Tuple containing:
+            - np.ndarray [surge, pitch, yaw]: commanded values
+            - float: distance to target
+            - float: depth error
+        """
+        # Extract positions
+        dx = target_pos[0] - current_pos[0]  # x
+        dy = target_pos[1] - current_pos[1]  # y
+        
+        # Horizontal plane calculations
+        horizontal_distance = np.sqrt(dx**2 + dy**2)
+        desired_yaw = np.arctan2(dy, dx)
+        yaw_error = self.normalize_angle(desired_yaw - current_pos[3])  # current_pos[3] is yaw
+        
+        # Vertical plane calculations
+        depth_error = target_pos[2] - current_pos[2]  # z
+        desired_pitch = self.compute_pitch_command(depth_error, horizontal_distance)
+        
+        # Compute desired surge velocity
+        desired_surge = self.compute_desired_speed(yaw_error, horizontal_distance)
+
+        # Return commands as numpy array
+        commands = np.array([desired_surge, desired_pitch, desired_yaw])
+        return commands, horizontal_distance, depth_error
```

Comment:
it would be nice if you have some sort of reference of where you found the equations. In the end, this will be in the README.md for the package, but for now it would be nice if the references were in comments in the code.